### PR TITLE
dApp -> WalletConnect -> ProviderService flow with v1 support

### PR DESCRIPTION
**Pull request by eladrof**
Closes https://github.com/tallyhowallet/extension/issues/2687
Closes https://github.com/tallyhowallet/extension/issues/2751
Closes https://github.com/tallyhowallet/extension/issues/2772
Partially involves https://github.com/tallyhowallet/extension/issues/2752

This PR enables full DApp to Wallet connection flow via WalletConnect for both protocol versions v1 and v2.

### To Test
- [ ] Enable connection button injection in [wallet-connection-handlers.ts](https://github.com/tallyhowallet/extension/blob/487f15a013c0229e19353d3ffad1f03db5cd68e9/window-provider/wallet-connection-handlers.ts#L503)
- [ ] Navigate to dApp of your choice that supports WalletConnect, such as [PancakeSwap](https://pancakeswap.finance/)
- [ ] Choose to connect via the WalletConnect option which would present a QR code screen with and injected button saying "Here be doggos"
- [ ] Click "Here be doggos" to initiate Wallet <-> DApp connection dialog, approve the connection
- [ ] Once connected, test the feature by performing any action on the DApp that requires signing. This should prompt the wallet to display a corresponding dialog prompt

Latest build: [extension-builds-2795](https://github.com/tallyhowallet/extension/suites/10669357566/artifacts/533530414) (as of Mon, 30 Jan 2023 18:58:50 GMT).

### CODE REVIEW

File: window-provider/wallet-connection-handlers.ts [hunk 1]
```diff
@@ -469,4 +497,8 @@ export default function monitorForWalletConnectionPrompts(): void {
       observeMutations(hostnameToHandler[hostname])
     }
   })
+
+  // Commenting this out for now, because we don't have a way to put this behind a feature flag
+  // SUPPORT_WALLET_CONNECT flag
+  // observeMutations(addTallyButtonForWalletConnectModal)
```
Comment (hyphenized) [inline window-provider/wallet-connection-handlers.ts:503]: Let's read the feature flag directly for now

```suggestion
  if (process.env.SUPPORT_WALLET_CONNECT === "true")
    observeMutations(addTallyButtonForWalletConnectModal)
```

File: background/services/wallet-connect/eip155-request-utils.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
+import { SignClientTypes } from "@walletconnect/types"
+import {
+  EIP1193Error,
+  EIP1193_ERROR_CODES,
+} from "@tallyho/provider-bridge-shared"
+import {
+  formatJsonRpcError,
+  formatJsonRpcResult,
+  JsonRpcError,
+  JsonRpcResult,
```
Comment (hyphenized) [inline background/services/wallet-connect/eip155-request-utils.ts:10]: We can remove this dependency in favour of using plain objects for results/errors

**Review by hyphenized — CHANGES_REQUESTED** [review]
Looks good! Left a couple of comments, please also fix the linting issues. Tested v1 and v2 flows, injected button does not show up in some modals but connecting manually still works

File: background/services/wallet-connect/legacy-sign-client-helper.ts [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+export function createLegacySignClient(
+  uri?: string,
+  sessionProposalListener?: SessionProposalListener,
+  sessionRequestListener?: SessionRequestListener
+): void {
+  // If URI is passed always create a new session,
+  // otherwise fall back to cached session if client isn't already instantiated.
+  if (uri) {
+    deleteCachedLegacySession()
+    legacySignClient = new LegacySignClient({ uri })
```
[the remark this answers is not in the document]
Comment (eladrof) [reply background/services/wallet-connect/legacy-sign-client-helper.ts:62]: This is the v1 client
It differs from v2 in lifecycle and functionality so we have no choice but to create and manage both v1 and v2 sign clients separately. We do however end up routing their outputs into a shared pipeline so at least there's that

**Review by hyphenized — APPROVED** [review]
Looks great! Excited to ship this soon!

**Comment by puppycodes** [comment]
Additional issue needed for WalletConnect QA

**Pull request merged**